docs/code_standards.rst

```diff
@@ -40,12 +43,33 @@ Docstrings
 Strings
 
 * Use string methods instead of the string module.
-* Use the :code:`f''` f-string syntax to `properly format strings`_::
-
-    formatted_string = f'I {key} a {value}.'
+* Use f-strings for string formatting whenever possible.
+* Use the string's format method when an f-string would become too complex.
+* When using format, leave out indices, unless absolutely necessary (variable re-use).
+
+Examples ::code::
+
+    simple_example = f'This is a {simple_var} example'
+    moderate_example = f'{some_var}: {some_func(arg="foo")}'
+    multi_line_example = f'You can also do {"larger"} '
+        'f-strings. Only using the f notation when a '
+        f'string requires it like this one: {some_func()}'
+
+    complex_formatting = 'This {} contains both {} and {}'.format(
+        'string',
+        '"layered"',
+        ' '.join(['a', 'non-trivial list', 'converted to', 'a', 'string'])
+    )
+    indices_needed = (
+        'this {var1} is a {var1} that has the {var2} {var1} {var3}'.format(
+            var1='sentence', var2='word', var3='multiple times'
+        )
```

Comment:
why not 
```
'this {0} is a {0} that has the {1} {0} {2}'.format('sentence', 'word', 'multiple times')
```

Reply:
Would people prefer using indices of kwargs here?

Reply:
The indices works well (look up) when we have less number of arguments to format but when the arguments count is high the kwargs works better.

If the number of code lines doesnt matter. then I would go with kwargs.

Reply:
I guess we won't see it very often, maybe we can leave it to contributor's discretion.